Products/ZenUI3/browser/resources/js/zenoss/itinfrastructure.js

```diff
@@ -117,20 +117,17 @@ Ext.onReady(function () {
                 component.store.load({
                     scope: this,
                     callback: function (records, operation, success) {
-                        var selnode = getSelectionModel().getSelectedNode();
-                        var isclass = selnode.data.uid.startswith('/zport/dmd/Devices');
-
-                        if (selnode.data.uid === "/zport/dmd/Devices" || !isclass) {
-                            //root node doesn't have a path attr
-                            this.setValue(this.store.first());
-                        }
-                        else if (isclass) {
-                            var path = selnode.data.path;
-                            path = path.replace(/^Devices/, '');
-                            if (this.findRecordByValue(path)) {
-                                this.setValue(path);
-                            }
+                        // no needs to take care of "CSE_VIRTUAL_ROOT" in devices uid's
+                        // simply use device path and strip everithing with "Devices" from it;
+                        var selnode = getSelectionModel().getSelectedNode(),
+                            path = selnode && selnode.data.path,
```

Comment:
The comment above says that the root node doesn't have the `path` attr. How doesn't this blow up?

Reply:
Each ExtJs tree has one main root node that is invisible for users and handle all other nodes.
It looks like this
    `root node
         device node
             AWS node
             Azure node
.......................................`
On store callback we select same device class in combo as in tree.
Before my changes it check if selected node is "Device" root or hidden (main root) of tree if so we select first device class otherwise we strip selected node path and search in combo store same.
I simplify it to use only selected node path and to search by it.